### ISSUE TITLE
Options would be clearer in as an options object parameter

### ISSUE THREAD

**Issue by pimterry** [konsumer/rawprotoparse#1]
Not by any means urgent, and it'd be a breaking change, but I think it'd make sense to move to an options object, changing:

```js
rawprotoparse(buffer, 'myFieldName', 'auto', false, getVal)
```

to

```js
rawprotoparse(buffer, {
  prefix: 'myFieldName',
  stringMode: 'auto',
  arrayMode: false,
  valueHandler: getVal
});
```

It'd make it much easier in the common case where you only want one of those options, and it's much clearer with the option names made explicit at call time.

**Comment by konsumer** [comment]
I am ok with this change. I am not sure how this might interact with #2. I would like to document the input options, but actively dislike typescript (but also am ok with annotating things, if it helps people who like it, though.) Currently, to address #1 , I added jsdoc for the params. What would this look like in a way that doesn't require all typescript tooling, but benefits all users?

```js
/*
 * @param {Uint8Array|Buffer} buffer The buffer of binary protobuf to parse
 * @param {string} [prefix='f'] A string-prefix to use for outputting objects. for example field id 1 will be "f1" if the prefix is "f"
 * @param {string} [stringMode='auto'] How to handle LEN fields, which can be byte-buffers, sub-messages, or strings. Could be "auto", "string", or "buffer".
 * @param {boolean} [arrayMode=false] arrayMode Should the output fields be fordced to arrays? This is to handle the idea that a field can have multiple values, which does not cleanly map to JSON. Normally, this will decide if it should be an array or not, but you can force arrays always, to keep it uniform.
 */
```

I guess i am unsure how how to document these options without using typescript-specific stuff (define a new `type`, which will require switching the language and the tools I am using, of my project.)

**Comment by pimterry** [comment]
Looks like the typing in jsdoc for options objects isn't too bad:

```jsdoc
 * @param {Object} [options]
 * @param {string} [options.prefix='f']
 * ...
```

I.e. add an `options` object param, and then rename every param within to `options.x`.

**Comment by konsumer** [comment]
That's not bad at all. i will add it.

**Comment by konsumer** [comment]
published to rawprotoparse@0.0.7
